### PR TITLE
Add clion debug scripts. Update README

### PR DESCRIPTION
**Pull request by PlagaMedicum**

### CODE REVIEW

File: clion_debug/Dockerfile.clion-cpp-env [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# CLion remote docker environment (How to build docker container, run and stop it)
+#
+# Build and run:
+#   docker build -t clion/remote-cpp-env:0.5 -f Dockerfile.clion-cpp-env .
```
Comment (MikhailSadovsky) [inline clion_debug/Dockerfile.clion-cpp-env:4]: Please change image name to "ostis/clion-remote-cpp-env" in all files

File: clion_debug/Dockerfile.clion-cpp-env [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#
+# stop:
+#   docker stop clion_remote_env
+# 
+# ssh credentials (test user):
+#   user@password 
+
+FROM ostis/ostis:0.5.0
+
+RUN apt-get update \
```
Comment (MikhailSadovsky) [inline clion_debug/Dockerfile.clion-cpp-env:16]: I think we don't need to install "build-essential, gcc, g++, clang, cmake, python" as it exists in ostis:0.5.0. Please re-check

File: clion_debug/Dockerfile.clion-cpp-env [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    echo 'PasswordAuthentication yes'; \
+    echo 'Subsystem sftp /usr/lib/openssh/sftp-server'; \
+    echo 'ChrootDirectory /'; \
+  ) > /etc/ssh/sshd_config_test_clion \
+  && mkdir /run/sshd
+
+RUN useradd -m user \
+  && yes password | passwd user
+
+RUN chmod ugo+rw /ostis/sc-machine
```
Comment (MikhailSadovsky) [inline clion_debug/Dockerfile.clion-cpp-env:41]: Please do all "chmod" commands in one RUN command

File: clion_debug/run_clion_debug.sh [hunk 4]
```diff
@@ -0,0 +1,122 @@
+#!/bin/sh
+
+PORT="8000"
+IMAGE="clion/remote-cpp-env"
+VERSION="0.5"
```
Comment (MikhailSadovsky) [inline clion_debug/run_clion_debug.sh:5]: Please use "0.5.0" version

File: clion_debug/build_clion_debug_image.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+IMAGE="clion/remote-cpp-env"
+VERSION="0.5"
```
Comment (MikhailSadovsky) [inline clion_debug/build_clion_debug_image.sh:4]: Please use "0.5.0" version

File: README.md [hunk 6]
```diff
@@ -50,10 +50,17 @@ Add ``` -v full_path_to_problem_solver_folder:/ostis/problem-solver``` to specif
 
 Note that C++ agents should be inside **problem-solver/cxx** folder, SCP agents should be inside **problem-solver/scp** folder.
 
-Example of usage:
+Example of the image usage:
 ```
-docker run -it -v /home/user01/test/kb:/ostis/kb -v /home/user01/test/problem-solver:/ostis/problem-solver -p 8000:8000 -p 8090:8090 ostis/ostis:0.6.0
+docker run -it -v ~/test/kb:/ostis/kb -v ~/test/problem-solver:/ostis/problem-solver -p 8000:8000 -p 8090:8090 ostis/ostis:0.5.0 sh ostis [OSTIS FLAGS]
 ```
+OSTIS FLAGS:
```
Comment (MikhailSadovsky) [inline README.md:57]: Please format all options to be viewed as a list with new line

File: README.md [hunk 7]
```diff
@@ -63,23 +70,26 @@ Run script has additional useful options comparing to Quickstart section. To run
...
+1. Run the script with needed options:
     ```bash
-    ./run.sh
+    ./run.sh [OPTIONS]
+    ```
+    See flags with
+    ```bash
+    run.sh --help
+    ```
+OPTIONS:
```
Comment (MikhailSadovsky) [inline README.md:82]: Please format all options to be viewed as a list with new line

File: README.md [hunk 8]
```diff
@@ -88,12 +98,68 @@ To build image locally you will need:
...
-3. Run build image script:
+1. Checkout to branch according to version you need
+1. Run build image script:
     ```bash
     ./build_image.sh
     ```
 
+## CLion Integration
+
+### Environement
```
Comment (MikhailSadovsky) [inline README.md:109]: Please update to "environment"

File: README.md [hunk 9]
```diff
@@ -88,12 +98,68 @@ To build image locally you will need:
...
+1. Clone sc-machine repo.
+```bash
+git clone --single-branch --branch 0.5.0 https://github.com/ShunkevichDV/sc-machine.git
+```
+1. Add problem-solver folder with your agents beside the sc-machine's folder.
```
Comment (MikhailSadovsky) [inline README.md:114]: Something with numeration here. It displays like first step, but should be second

File: README.md [hunk 10]
```diff
@@ -88,12 +98,68 @@ To build image locally you will need:
...
+![folders](./img/clion/folders.png) 
+1. Add `SET(CMAKE_BUILD_TYPE Debug)` at the end of `CMakeLists.txt` of your module.
```
Comment (MikhailSadovsky) [inline README.md:116]: I think we need to move this step later as exampleModule will not be displayed in CLion at this stage

File: README.md [hunk 11]
```diff
@@ -88,12 +98,68 @@ To build image locally you will need:
...
+    ```
+1. Run clion debug container with needed options:
+    ```bash
+    ./clion_debug/run_clion_debug.sh [OPTIONS]
+    ```
+    See flags with
+    ```bash
+    ./clion_debug/run_clion_debug.sh --help
+    ```
+OPTIONS:
```
Comment (MikhailSadovsky) [inline README.md:132]: Please format all options to be viewed as a list with new line

File: README.md [hunk 12]
```diff
@@ -88,12 +98,68 @@ To build image locally you will need:
...
+    `--help -h` - Print help message
+    `--port -p` - Set a custom port
+    `--app` - Set a custom path to the app directory(By default, it is expected, that inside the app you have all default directories for kb, problem-solver etc)
+    `--kb` - Set a custom path to kb directory
+    `--solver` - Set a custom path to problem-solvers deirectory
+    `--startflags --sf` - To set container startup flags(using `--build_kb --sc-web` by default). Usage: `--startflags "[OSTIS FLAGS]"`  
+1. Debug your code! (See project configuration below)
```
Comment (MikhailSadovsky) [inline README.md:139]: It should be step 3, but displays as 1

File: README.md [hunk 13]
```diff
@@ -88,12 +98,68 @@ To build image locally you will need:
...
+1. After finishing your work stop and remove debug container:
+    ```bash
+    ./clion_debug/stop_clion_debug.sh
+    ```
+
+### Configuring CLion
+1. Add toolchain in __Settings/Preferences | Build, Execution, Deployment | Toolchains__ adding new ssh connection.
```
Comment (MikhailSadovsky) [inline README.md:146]: Please specify default user/password

File: README.md [hunk 14]
```diff
@@ -88,12 +98,68 @@ To build image locally you will need:
...
+![toolchain config](./img/clion/toolchains.png) 
+![ssh config](./img/clion/ssh_config.png)
+1. Add mew CMake profile for container in __Settings/Preferences | Build, Execution, Deployment | CMake__.
```
Comment (MikhailSadovsky) [inline README.md:149]: Please fix typo to "new"

File: README.md (hunk 11 above)
Comment (PlagaMedicum) [reply README.md:132]: fix

File: README.md (hunk 7 above)
Comment (PlagaMedicum) [reply README.md:82]: fix

File: README.md (hunk 6 above)
Comment (PlagaMedicum) [reply README.md:57]: fix

File: clion_debug/Dockerfile.clion-cpp-env (hunk 2 above)
Comment (PlagaMedicum) [reply clion_debug/Dockerfile.clion-cpp-env:16]: fix

File: README.md (hunk 10 above)
Comment (PlagaMedicum) [reply README.md]: yes. But user can edit this at any stage in any environment

**Pull request merged**